src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  removeTodo: (value: number) => void,
+  setSelectedTodos: (value: number[]) => void,
+  onUpdate: (todoId: number, data: Partial<Todo>) => void,
+  selectedTodos: number[],
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  removeTodo,
+  setSelectedTodos,
+  onUpdate,
+  selectedTodos,
+}) => {
+  const handleRemove = () => {
+    removeTodo(todo.id);
+    setSelectedTodos([todo.id]);
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames(
+        'todo',
+        { completed: todo.completed },
+      )}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          onClick={() => onUpdate(todo.id, { completed: !todo.completed })}
```

Comment:
Create a separate method instead of creating an inline event handler.
Do the same all over the code when needed.